genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -102,7 +102,10 @@ protected void executePreActionValidation() {
     protected Events executeStateAction(
         final ExecutionContext executionContext
     ) {
+        log.info("Setting up job...");
+
         final String claimedJobId = executionContext.getClaimedJobId().get();
+        final JobSpecification jobSpecification = executionContext.getJobSpecification().get();
```

Comment:
Possible exception here if there is no `jobSpecification`. Perhaps you're making the assumption it's already there but you might want to use `orElseThrow()` and control what exception and message are thrown